backend/hitas/views/apartment.py

```diff
@@ -492,20 +493,40 @@ class ApartmentDetailSerializer(EnumSupportSerializerMixin, HitasModelSerializer
     improvements = ApartmentImprovementSerializer(source="*")
 
     conditions_of_sale = serializers.SerializerMethodField()
+    sell_by_date = serializers.SerializerMethodField()
 
-    def get_conditions_of_sale(self, instance: Apartment) -> list[dict[str, Any]]:
-        conditions_of_sale: list[ConditionOfSale] = []
+    @staticmethod
+    def get_conditions_of_sale(instance: Apartment) -> list[dict[str, Any]]:
+        conditions_of_sale: list[ConditionOfSale] = [
+            cos
+            for ownership in instance.ownerships.all()
+            for cos in chain(ownership.conditions_of_sale_new.all(), ownership.conditions_of_sale_old.all())
+        ]
+
+        return [ConditionOfSaleSerializer(condition_of_sale).data for condition_of_sale in conditions_of_sale]
+
+    @staticmethod
+    def get_sell_by_date(instance: Apartment) -> Optional[datetime.date]:
+        sell_by_dates: set[datetime.date] = set()
 
         for ownership in instance.ownerships.all():
-            conditions_of_sale_new = ownership.conditions_of_sale_new.all()
-            for cond in conditions_of_sale_new:
-                conditions_of_sale.append(cond)
+            for cos in chain(ownership.conditions_of_sale_new.all(), ownership.conditions_of_sale_old.all()):
+                if cos.fulfilled:
+                    continue
```

Comment:
Shouldn't this already not include fulfilled conditions of sale?
Alternatively we can just add a `.filter(deleted__isnull=True)` to the querysets to not include them

Reply:
The queryset does include fulfilled conditions of sale for some time after they are fulfilled [based on a setting](https://github.com/City-of-Helsinki/hitas/blob/master/backend/hitas/models/condition_of_sale.py#L71) since this endpoint is using `condition_of_sale_queryset()`. This is useful for showing recently fulfilled conditions of sale as per the design.

Also, we shouldn't filter them here further since that will invalidate the prefetch cache (I think).